src/core/core.datasetController.js

```diff
@@ -663,11 +664,34 @@ helpers.extend(DatasetController.prototype, {
 				minPositive = Math.min(minPositive, value);
 			}
 		}
-		return {
-			min: min,
-			max: max,
-			minPositive: minPositive
-		};
+
+		function _skip() {
+			item = data[i];
+			parsed = _parsed[i];
+			value = parsed[scale.id];
+			otherValue = parsed[otherScale.id];
+			return ((item && item.hidden) || isNaN(value) || otherMin > otherValue || otherMax < otherValue);
+		}
+
+		for (i = 0; i < ilen; ++i) {
+			if (_skip()) {
+				continue;
+			}
+			_compute();
+			if (sorted) {
+				break;
+			}
+		}
+		if (sorted) {
+			for (i = ilen - 1; i >= 0; --i) {
```

Comment:
Will this do the computation a second time if the data is not sorted? It seems like we might be able to make it more efficient and readble by checking just the first and last data point if sorted and then looping only once if not sorted

Reply:
No, this part gets the max value from the other end when sorted.
I preserved the logic that skipped or values that are out of chart are when other scale is limited by min/max are ignored here, even if the data is sorted.

So the sorted case, we start from 0 and continue until a point that is not skipped is found. Then the loop breaks and we do the same from `ilen - 1` backwards.

In a usual (sorted) case, both loops are executed just once.

Reply:
I was asking about the unsorted case though

Reply:
Oh sorry. 2nd loop is only executed if sorted, so the unsorted case stays the same.

Reply:
Oh, sorry. I must have read it wrong the first time